d2ir/d2ir.go

```diff
@@ -687,8 +690,9 @@ func (m *Map) ensureField(i int, kp *d2ast.KeyPath, refctx *RefContext) (*Field,
 	}
 
 	f := &Field{
-		parent: m,
-		Name:   head,
+		parent:     m,
+		Name:       headString,
+		IsUnquoted: headUnquoted,
```

Comment:
You also need to check that there are no escapes. Use `len(headString) == len(head.(*d2ast.UnquotedString).RawString)`. i.e there are no escapes if the length of the string value is equal to it's length in the AST.

Reply:
IsUnquoted could be confusing then, but i can't think of a word that encapsulates "is unquoted or is escaped". Raw already means something else. will leave as IsUnquoted for now unless u have a word

Reply:
Yea just add a comment on `IsUnquoted`. I have no better word in mind.